tests/end2end/features/test_editor_bdd.py

```diff
@@ -64,3 +66,33 @@ def set_up_editor(quteproc, server, tmpdir, text):
 def set_up_editor_empty(quteproc, server, tmpdir):
     """Set up editor.command to a small python script inserting empty text."""
     set_up_editor(quteproc, server, tmpdir, "")
+
+
+@bdd.when(bdd.parsers.parse('I set up a fake editor that waits'))
+def set_up_editor_wait(quteproc, server, tmpdir):
+    """Set up editor.command to a small python script inserting a text."""
+    pidfile = tmpdir / 'editor_pid'
+    script = tmpdir / 'script.py'
+    script.write(textwrap.dedent("""
+        import os
+        import sys
+        import time
+        import signal
+
+        with open('{pidfile}', 'w') as f:
+            f.write(str(os.getpid()))
+
+        signal.signal(signal.SIGUSR1, lambda s, f: sys.exit(0))
+
+        time.sleep(100)
+    """.format(pidfile=pidfile)))
+    editor = json.dumps([sys.executable, str(script), '{}'])
+    quteproc.set_setting('editor.command', editor)
+
+
+@bdd.when(bdd.parsers.parse('I kill the waiting editor'))
+def kill_editor_wait(quteproc, server, tmpdir):
+    """Kill the waiting editor."""
+    pidfile = tmpdir / 'editor_pid'
+    pid = int(pidfile.read())
+    os.kill(pid, signal.SIGUSR1)
```

Comment:
Dang, pylint correctly points out that this won't work on Windows 😆 I wonder whether using `SIGTERM` instead (and handling that in the process) would work on Windows as well.